napari/layers/utils/color_manager.py

```diff
@@ -405,9 +405,10 @@ def _update_current_properties(
             current_property_name = self.color_properties.name
             current_property_values = self.color_properties.values
             if current_property_name in current_properties:
-                new_current_value = np.squeeze(
-                    current_properties[current_property_name]
-                )
+                new_current_value = current_properties[current_property_name][
```

Comment:
I stepped through the test and found this to be `'p'` - is that really what we want?

I assume that after running `layer.current_properties = {'annotation': 'paw'}` we would have `layer._edge.color_properties.current_value == 'paw'` (at least for the edge `ColorManager`).

Reply:
Thanks for catching this, @andy-sweet . You are correct. This is actually a bug in the test. I should have written `layer.current_properties = {'annotation': np.array(['paw'])}`. I have fixed it and added `assert layer._edge.color_properties.current_value == 'paw'` line to reflect this.

This makes me realize we should probably allow users to pass `layer.current_properties` as a dictionary with values as a str/float/int/bool, as that's a pretty natural thing to do. I've added a function to the  `Points.current_properties` and `Shapes.current_properties` setters that will coerce the input into the expected dictionary with 1D numpy array values.